### ISSUE TITLE
Improve supported gate set documentation for hardware providers

### ISSUE THREAD

**Issue by tcNickolas** [MicrosoftDocs/quantum-docs#112]
Several things can be confusing for somebody who's trying to figure out the supported gate set:
1. Native gate set is briefly documented in Honeywell provider spec but not in IonQ, for IonQ's set one has to dig into "Best practices" page.
2. What happens if Q# code uses gates that are not native for the provider? I'm assuming that (in the spirit of Q# being a high-level language) those gates are compiled into native gate set, it would be nice to mention this.
3. The abbreviation SPAM is decoded only in Honeywell provider spec, and I couldn't find an explanation of what "state preparation and measurement" operation actually is.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: d6b0c1b5-1576-373a-2797-b2dda7963ca6
* Version Independent ID: 999d0181-a2ef-bc76-0964-0c55244d8488
* Content: [IonQ provider - Azure Quantum ](https://docs.microsoft.com/en-us/azure/quantum/provider-ionq)
* Content Source: [articles/provider-ionq.md](https://github.com/MicrosoftDocs/quantum-docs/blob/main/articles/provider-ionq.md)
* Service: **azure-quantum**
* Sub-service: **computing**
* GitHub Login: @KittyYeungQ
* Microsoft Alias: **kitty**

**Comment by dime10** [comment]
Thank you for providing these suggestions on how to improve the hardware documentation.

@geduardo Are you able to look into this?

**Comment by geduardo** [comment]
This is a tricky discussion. In general, it was agreed that providers should provide all the target-specific documentation for their targets and not us. But I agree that there are many points of intersection between the target-specific documentation and the connection with Q# in which we may need to connect the dots. At least what I would say that we should be consistent and provide similar information for all targets and all providers. I think we are not succeeding at that at the moment.

I would like @Mobius5150 to chime in and give us his point of view since we have discussed this in the past.

**Comment by dphansen** [comment]
@Mobius5150 Would you be able to have a look at this GitHub issue? Thank you.

**Comment by Mobius5150** [comment]
I agree that it would be good to document the available gate set for providers, and the compilation process as it pertains to how things get compiled down. This is information that is contained within our Target Profile in the Q# compiler so it doesn't seem unreasonable for us to document.

I'll check in with Honeywell re: SPAM.
